src/walletConnect/request.test.ts

```diff
@@ -106,33 +120,101 @@ describe(handleRequest, () => {
+      .call([mockViemWallet, 'signTypedData'], mockTypedData)
       .run()
   })
 
   it('supports eth_signTypedData_v4', async () => {
     await expectSaga(handleRequest, signTypedDataV4Request)
-      .provide([[call(getWallet), mockWallet]])
+      .provide([[matchers.call.fn(getViemWallet), mockViemWallet]])
       .withState(state)
-      .call(unlockAccount, '0xwallet')
-      .call([mockWallet, 'signTypedData'], '0xwallet', { message: 'Some typed data' })
+      .call([mockViemWallet, 'signTypedData'], mockTypedData)
       .run()
   })
 
   describe('eth_signTransaction', () => {
-    describe('transaction normalization', () => {
+    // TODO: keep only Viem branch after feeCurrecny estimation is ready
+    describe('transaction normalization (viem)', () => {
+      beforeAll(() => {
+        jest.mocked(getFeatureGate).mockReturnValue(true)
+      })
+
+      it('ensures `gasLimit` value is moved to the `gas` parameter', async () => {
+        await expectSaga(handleRequest, {
+          method: SupportedActions.eth_signTransaction,
+          params: [{ from: '0xTEST', data: '0xABC', gasLimit: '0x5208' }],
+        })
+          .provide([
+            [matchers.call.fn(getViemWallet), mockViemWallet],
+            [matchers.call.fn(getTransactionCount), mockTransactionCount],
+            [matchers.call.fn(estimateFeesPerGas), mockEstimateFeePerGas],
+          ])
+          .withState(state)
+          .call(formatTransaction, { from: '0xTEST', data: '0xABC', gas: '0x5208' })
+          .run()
+      })
+
+      it('ensures `gasPrice` is stripped away before preparing transaction request', async () => {
+        await expectSaga(handleRequest, {
+          method: SupportedActions.eth_signTransaction,
+          params: [{ from: '0xTEST', data: '0xABC', gasPrice: '0x5208' }],
+        })
+          .provide([
+            [matchers.call.fn(getViemWallet), mockViemWallet],
+            [matchers.call.fn(getTransactionCount), mockTransactionCount],
+            [matchers.call.fn(estimateFeesPerGas), mockEstimateFeePerGas],
+          ])
+          .withState(state)
+          .call(formatTransaction, { from: '0xTEST', data: '0xABC' })
+          .run()
+      })
+
+      it('ensures normalization is skipped when __skip_normalization is set', async () => {
+        await expectSaga(handleRequest, {
+          method: SupportedActions.eth_signTransaction,
+          params: [
+            {
+              from: '0xTEST',
+              data: '0xABC',
+              gasLimit: '0x5208',
+              gasPrice: '0x5208',
+              __skip_normalization: true,
+            },
+          ],
+        })
+          .provide([
+            [matchers.call.fn(getViemWallet), mockViemWallet],
+            [matchers.call.fn(getTransactionCount), mockTransactionCount],
+            [matchers.call.fn(estimateFeesPerGas), mockEstimateFeePerGas],
+          ])
+          .withState(state)
+          .call(formatTransaction, {
```

Comment:
is it possible to asset that the mockViemWallet signTransaction method was called with these fields instead? not sure if it's possible with the big ints :/ but it would be nice since that's the end result (the current test could still pass if we incorrectly changed parts of the function after `formatTransaction` was called correctly)

Reply:
I've planned this test block with a limited scope, just to ensure that normalization works correctly when it should.

    Viems's `signTransaction()` must only be called with `bigint` values. Calling it with string hex values will result in an error. That's why currently `formatTransaction` must precede any further action with transaction request. Ideally, Viem should handle formatting itself under the hood.
    
    I'm doubtful that checking the whole code of `sign_transaction` action end to end can add any value, because most of the functions inside are mocked -- they are from Viem or other helpers. The only place where we are executing our own meaningful code is normalization, which is covered here.

    I think the real e2e test are better suited to check that we still can sign a valid transaction.

    What do you think?

Reply:
yes i agree with all of that :)